src/main/java/me/x150/renderer/objfile/ObjFile.java

```
@@ -157,14 +163,85 @@ private void bake() {
 		baked = true;
 	}
 
+
+	protected Vector3f calculateSunPosition(long worldTime) {
+
+		double timeFraction = worldTime % 24000.0 / 24000.0;
+		double angle = ((timeFraction) * 2.0 * Math.PI);
+
+		// Calculate sun X and Y position using cosine and sine to create the unit circle.
+		// Reverse Y since in most systems, positive Y goes up.
+		float sunX = (float) Math.cos(angle);
+		float sunY = (float) Math.sin(angle);
+
+		// The sun doesn't move in the Z direction so Z is 0.
+		float sunZ = 0.0f;
+
+		return new Vector3f(sunX, sunY, sunZ);
+	}
+
+
+	/**
+	 * You can provide your own shader
+	 * Overridable, is called in draw()
+	 *
+	 * @return shader used in draw
+	 */
+	protected ShaderProgram getNormalLitShader()
+	{
+		return ShaderManager.OBJ_SHADER.getProgram();
+	}
+
 	/**
 	 * Draws this ObjFile. Calls {@link #bake()} if necessary.
+	 * Automatically calculates light level at only the origin applied to the entire object.
```

Wording is a little weird. suggestion: `Applies correct main world lighting to the drawn object, calculated from the object's {@code origin}`

---

should also probably note that if no world is present, this method does nothing. maybe throw an exception instead, or let the user provide the world?